src/reducers.js

```diff
@@ -109,7 +109,7 @@ function interfaces(state, action) {
         }
 
         const updatedIface = Object.assign({}, state[index], iface);
-        return replaceResource({ state, updatedIface, index });
+        return replaceResource({ state, updatedResource: updatedIface, index });
```

Comment:
How does this exactly affect a bug which only happens after shutting down the libvirtd.service? Are these function not also called initially when the page is first loaded? Just trying to understand the relation to the bug.

Reply:
They are also called when you stop libvirtd just with delay polling:
https://github.com/cockpit-project/cockpit-machines/blob/main/src/libvirt-dbus.js#L1138

Reply:
Oh I see, so then they would be replaced by incorrect object/list?
The code itself seems good, so giving.

Reply:
Correct, that's what happened resulting in a JS exception.